### PR TITLE
RDS split updated with new integration tests [slow tests]

### PR DESCRIPTION
**Pull request by michael-dev2rights**
##### SUMMARY

This is an updated version of https://github.com/ansible/ansible/pull/19772 from Willthames which contains fixes as well as widely extended integration test cases which have been moved into the new integration testing framework and so would work within CI.  

> Split the existing rds module into three separate modules, and make the modules idempotent (remove the command argument).
>  The new modules are expected to be almost entirely backward compatible with the old module, other than the command change - it should be easy to update existing code to use the new modules.
> The new modules are entirely boto3 - support for boto has been dropped for the new modules.

Willthames stated that he's not going to have much time to work on this (though hopefully he'll be able provide a little support and e.g. review the code) so I'm now proposing to put this forward myself as a separate PR

##### ISSUE TYPE

* New Module Pull Request

##### COMPONENT NAME

rds_instance
rds_instance_facts (see also https://github.com/ansible/ansible/pull/21884 - will rebase if that merges)
rds_snapshot
rds_snapshot_facts (see also https://github.com/ansible/ansible/pull/21884 - will rebase if that merges)

##### ANSIBLE VERSION

```
ansible 2.4.0 (mdd_rds_split_fix_new 279738a32d) last updated 2017/06/13 10:55:20 (GMT +100)
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/mikedd/dev/ansible/ansible-add-modules/library']
  ansible python module location = /home/mikedd/dev/ansible/ansible-add-modules/lib/ansible
  executable location = /home/mikedd/dev/ansible/ansible-add-modules/bin/ansible
  python version = 2.7.12 (default, Jul  1 2016, 15:12:24) [GCC 5.4.0 20160609]
```

##### ADDITIONAL INFORMATION

For examples of using the module please see the included integration tests for now.

##### KNOWN ISSUES

* (maybe fixed?) Some modifications don't really work as expected - #19751 might apply equally here
Test suite takes a really long time and state transitions are very slow, however the run has been completed.

### CODE REVIEW

**Comment by michael-dev2rights** [comment]
In order to allow sensible test cases to succeed, this patch currently incorporates the rds_facts modules which are included in https://github.com/ansible/ansible/pull/21884.  I assume that that PR should be merged first at which point the two fact modules will be dropped and this PR becomes a simple replacement for rds.

**Comment by michael-dev2rights** [comment]
@mattclay - this one has permission problems as follows:
```
An error occurred (AccessDenied) when calling the DescribeDBInstances operation: User: arn:aws:sts::966509639900:assumed-role/ansible-core-ci-test-prod/prod=shippable=ansible=ansible=26009.40 is not authorized to perform: rds:DescribeDBInstances on resource: arn:aws:rds:us-east-1:966509639900:db:shippable-26009-40-rds-restore\n", 
```

**Comment by mattclay** [comment]
@michael-dev2rights The integration test target names need to match the names of the modules they're testing.

**Comment by michael-dev2rights** [comment]
@mattclay what's the advice in the cases where:

- testing two or modules together e.g.  
-- postgresql_user module in the case when its run against an rds_instance created database
-- compatiblitiy of rds_instance and rds_fact with the old rds module
- making a large set of different test cases which can benefit from separation from the main ones
-- slow RDS tests
-- compatibility tests as above.  
- the large complex multi-role tests like the later ec2 ones

It seems to me that these don't best fit into the case of integration test name always matches module name?

**Comment by michael-dev2rights** [comment]
@mattclay back to permission errors, could you please add the needed permissions?
` An error occurred (AccessDenied) when calling the DescribeDBInstances operation: User: arn:aws:sts::966509639900:assumed-role/ansible-core-ci-test-prod/prod=shippable=ansible=ansible=27021.40 is not authorized to perform: rds:DescribeDBInstances on resource: arn:aws:rds:us-east-1:966509639900:db:shippable-27021-40-rds\n`

File: test/integration/targets/rds_instance/README.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+Role Name
```
Comment (mattclay) [inline test/integration/targets/rds_instance/README.md:1]: This file appears to have been included unintentionally.

**Comment by michael-dev2rights** [comment]
I have now split this PR into separate PRs for each python module plus one each for module utils and integration tests.  I also created new separate PRs for the rds_*_facts modules referenced above.  

* rds module utils (this)  https://github.com/ansible/ansible/pull/26594 
* rds_instance_facts https://github.com/ansible/ansible/pull/26598
* rds_snapshot_facts https://github.com/ansible/ansible/pull/26599
* rds_instance https://github.com/ansible/ansible/pull/26602
* rds_snapshot https://github.com/ansible/ansible/pull/26604
* rds integration tests https://github.com/ansible/ansible/pull/26606

I believe that all comments in this PR have been taken into account so I'm closing it.  Please discuss in the PRs I just listed.

**Pull request closed without merging**